src/main/java/uk/gov/pay/directdebit/payments/params/PaymentViewSearchParams.java

```diff
@@ -137,6 +171,9 @@ public String generateQuery() {
         if (amount != null) {
             sb.append(" AND t.amount = :" + AMOUNT_FIELD);
         }
+        if (isNotBlank(state) && externalPaymentToInternalStateQueryMap.containsKey(state)) {
+            sb.append(externalPaymentToInternalStateQueryMap.get(state));
```

Comment:
here I would do: 
`sb.append(format(" AND t.state = '%s'", externalPaymentToInternalStateQueryMap.get(state)));`

I think it is clearer what's going on, and it is similar to the other fields